### ISSUE TITLE
Panic caused by runtime.gopark

### ISSUE THREAD

**Issue by syllith** [golang/go#57728]
Go version: 1.19.4
Reproduces with latest: Yes
PC info: Windows / amd64
Problem: I have a program that utilizes the net/http package to make thousands of network requests. This program is intended to be run for long periods of time and somewhere between 5 - 20 hours of running, it will eventually hit an odd panic regarding the runtime.gopark function. Here's what I am doing:

```
var client http.Client
var wg[] sync.WaitGroup

func main() {
    for i: = 0; i < 1000; i++ {
        wg = append(wg, sync.WaitGroup {})
        go loopPremium(i)
    }
}

func loopPremium(index int) {
    for {
        randomUri: = globals.ProductLists[0].Premium[rand.Intn(len(globals.ProductLists[0].Premium))]
        scrape(index, randomUri)
    }
}

func scrape(index int, uri string) {
    response: = make(chan http.Response, 10)
    for i: = 0; i < 10; i++{
        wg[index].Add(1)
        go multiRequest(response, index, uri)
    }
    wg[index].Wait()
    close(response)

    var passedRes http.Response
    for res: = range response {
        if res.StatusCode == 200 {
            passedRes = res
            break
        } else if res.Body != nil {
            res.Body.Close()
        }
    }

    if passedRes.StatusCode != 200 {
        // No successful response, return
        return
    }

    // Continue with rest of code
}

func multiRequest(c chan http.Response, index int, uri string) {
    defer wg[index].Done()

    req, _: = http.NewRequest("GET", uri, nil)
    resp, _: = client.Do(req)

    // If we lose connection, response will be nil. If this is the case, return an empty response instead
    if resp != nil {
        c < - * resp
    } else {
        c < -http.Response { }
    }
}
```

Explanation: The idea here is that I have a website that may or may not give me a response. What I want is to create an array of 1,000 worker groups and each one will call the `scrape` function that will pick a random URL and send it 10 requests back to back. If any of them finish successfully, continue with the code. Otherwise return. This is working as intended, however I am getting [this](https://skydrive.digi-safe.co/files/Documents/error.txt) call stack reported back to me as well as a crash after a few hours of running.

I am unable to tell if this is a bug or if I simply have a bad implementation. I realize that this is kind of a crazy amount of network requests, but it does work for hours on end, so in theory I should be able to do this. The reason I'm posting it as a possible bug is because I've already got feedback from other gophers and they don't see why this would be happening either. The panic is occurring in the Go runtime, so I think it's reasonable to think it might possibly be a bug. I suspect the go scheduler or http.client is not managing its resources properly? Does anybody have any ideas what could be going on here and how I can prevent these crashes? Thanks

**Comment by seankhliao** [comment]
i don't see anything pointing to the runtime, importantly, the top of the output appears to be cut off. is it possible to get the complete output?

**Comment by syllith** [comment]
Of course, here is another screenshot I managed to get the other day. Perhaps I'm misinterpreting it, but the call stack mentions runtme.gopark a lot, so I assumed it had something to do with that.

![image](https://user-images.githubusercontent.com/42996651/211642058-ec8a65cf-9346-4a7f-ae9d-134b970521a9.png)

**Comment by seankhliao** [comment]
That just means a goroutine is parked, waiting for an event (in your case, channel communication). It doesn't mean it's the cause of the crash.

**Comment by syllith** [comment]
I see. I'm sorry but this problem has me quite lost and I've spent a long time trying to identify it. Is there anything you can see that I should be investigating?

**Comment by seankhliao** [comment]
no, the important part is the beginning, which will tell you what sort of issue you're dealing with, but you've left that out.

closing as this doesn't appear to be in issue with go itself.

Unlike many projects, the Go project does not use GitHub Issues for general discussion or asking questions. GitHub Issues are used for tracking bugs and proposals only.

For questions please refer to https://github.com/golang/go/wiki/Questions

**Comment by seankhliao** [comment]
there's a header you've left out that comes before the stack traces.

Though the last goroutine dumps points to the program running out of memory.

**Comment by syllith** [comment]
Ah I see. Sorry, the screenshots are hard to get since the error doesn't happen very often. But, I discovered something just now and it seems you're right. Looking through the call stack, I noticed this line here:
![image](https://user-images.githubusercontent.com/42996651/211652651-ec4dd642-208d-4afb-b124-8e940d2b5ef1.png)

After looking at the file and line number, this is what it says:
![image](https://user-images.githubusercontent.com/42996651/211652710-24f9d5ed-8f81-4c37-9d62-7c67f3143573.png)

I find it interesting though that I couldn't find the text "out of memory" mentioned anywhere in the console log, not sure why that might be, but either way it seems I have a memory leak. That should be enough for me to work with, thank you very much for your time. Have a great day.
